### ISSUE TITLE
The iwlwifi_microcode_crash test only checks the journal log for the target kernel that is under test.

### ISSUE THREAD

**Issue by kevinyehk** [canonical/checkbox#1461]
### Enhancement Proposal

The `iwlwifi_microcode_crash` checks all the journal logs on the DUT, but when the DUT is noprovision, it might cause false alarm, as the microcode carsh event could have occurred in previous kernel.  
Therefore, I think we should modify the test to only check the target kernel/snap that is under test.

**Comment by anthonywong** [comment]
Can we stop reporting test failures like https://bugs.launchpad.net/ubuntu/+source/linux/+bug/2077537 until this bug is fixed?

**Comment by KaiChuan-Hsieh** [comment]
@kevinyehk Do you think clear old journal log before running the SRU test plan can avoid the situation?

**Comment by kevinyehk** [comment]
@KaiChuan-Hsieh yes, clear old log in the beginning of entire test session should be OK.

**Comment by boukeas** [comment]
I vacuumed the journal on `hp-elitebook-650-156-inch-g10-c31123`, which is a `noprovision` device, removing anything older than two days old, and reran the SRU test in which only the `iwlwifi_microcode_crash` test was [previously failing](https://certification.canonical.com/hardware/202301-31123/submission/393701/test-results/fail/). I can confirm [there were no failures](https://certification.canonical.com/hardware/202301-31123/submission/396673/).

**Comment by kevinyehk** [comment]
if we check the kernel version at the first line in the log matches the one running now instead of cleaning all the journal log. maybe it's a better way? 
```
    bootidx=$(echo "$line" | cut -d " " -f1)
    kernel=$(journalctl -k -b "$bootidx" | grep -oP 'Linux version \d+\.\d+\.\d+-\d+-.*' | cut -d " " -f 3 )
    if [ "$kernel" = "$(uname -r)" ]; then
            echo "checking microcode error...."
            if journalctl -k -b "$bootidx" | grep -q "Microcode SW error detected"; then
                    echo "Boot $line, Microcode SW error detected"
                    exit 1
            fi
    else
            echo "skip...."
    fi
```
